doc/tutorial.rst

```diff
@@ -435,36 +443,39 @@ In addition to constructing graphs node-by-node or edge-by-edge, they
 can also be generated by
 
 1. Applying classic graph operations, such as:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. autosummary::
 
-    subgraph          - induced subgraph view of G on nodes in nbunch
-    union             - graph union
-    disjoint_union    - graph union assuming all nodes are different
-    cartesian_product - return Cartesian product graph
-    compose           - combine graphs identifying nodes common to both
-    complement        - graph complement
-    create_empty_copy - return an empty copy of the same graph class
-    to_undirected     - return an undirected representation of G
-    to_directed       - return a directed representation of G
+    ~networkx.classes.function.subgraph
+    ~networkx.algorithms.operators.binary.union
+    ~networkx.algorithms.operators.binary.disjoint_union
+    ~networkx.algorithms.operators.product.cartesian_product
+    ~networkx.algorithms.operators.binary.compose
+    ~networkx.algorithms.operators.unary.complement
+    ~networkx.classes.function.create_empty_copy
+    ~networkx.classes.function.to_undirected
+    ~networkx.classes.function.to_directed
```

Comment:
CAn we avoid providing the entire path to functions?  I know this is a bigger question that this PR, but our reformatted docs theme makes all the doc pages title include the source entire code path to the function appear instead of promoting our flatnamespace.  Since that change we have been getting more PR code that relies on the source path rather than the flat namespace. This makes our code less consistent and more intimidating to new users who feel that they have to learn the source code tree rather than just learning the code they are using. I feel like adding the source code path to these functions continues this motion away from a flat namespace. If we've decided to move to a tree namespace, then much more than this list should be added to the tutorial.

If this is added so that the documentation can find a pointer to the docs for these functions, then lets find a way to make the link appear without listing the entire source code path.

Or... we can talk about whether to split up the flat namespace and teach the users the source code tree.

Reply:
The good news is that the `~` prevents the full path from being rendered in the html. Here's the section from the rendered docs: https://2544-890377-gh.circle-artifacts.com/0/doc/build/html/tutorial.html#applying-classic-graph-operations-such-as. Note that the html output is the [same as before](https://networkx.org/documentation/latest/tutorial.html#graph-generators-and-graph-operations) (i.e. the paths aren't shown) except the links to the corresponding docstrings now work.

So the tradeoff for this case is: less clear (i.e. full path) rst for better html (w/links) output. Unfortunately this is a common pattern with rst, which is not (IMO) a particularly readable markup, especially in the case of sphinx-generated internal links.

More generally I agree w/ the sentiment in #4855 and the figuring out how to improve the interaction with the theme to get rid of the full-paths problem in e.g. the sidebar. I have some ideas how to do this (to be discussed elsewhere) but no silver-bullets.